Classes/Command/CommandCommand.php

```diff
@@ -0,0 +1,168 @@
+                $defaultCommandClassName,
+            );
+
+            if ($commandClassName === '') {
+                $io->error('Class name can not be empty.');
+                $validCommandClassName = false;
+            } elseif (preg_match('/^\d/', $commandClassName)) {
+                $io->error('Class name should not start with a number.');
+                $defaultCommandClassName = $this->tryToCorrectCommandClassName($commandClassName);
+                $validCommandClassName = false;
+            } elseif (preg_match('/[^a-zA-Z0-9]/', $commandClassName)) {
+                $io->error('Class name contains invalid chars. Please provide just letters and numbers.');
+                $defaultCommandClassName = $this->tryToCorrectCommandClassName($commandClassName);
+                $validCommandClassName = false;
+            } elseif (preg_match('/^[A-Z][a-zA-Z0-9]+$/', $commandClassName) === 0) {
+                $io->error('Action must be written in UpperCamelCase like "DoSomethingCommand".');
+                $defaultCommandClassName = $this->tryToCorrectCommandClassName($commandClassName);
+                $validCommandClassName = false;
+            } elseif (!str_ends_with($commandClassName, 'Command')) {
+                $io->error('Class name must end with "Command".');
+                $defaultCommandClassName = $this->tryToCorrectCommandClassName($commandClassName);
+                $validCommandClassName = false;
+            } else {
+                $validCommandClassName = true;
+            }
+        } while (!$validCommandClassName);
+
+        return $commandClassName;
+    }
+
+    private function askForCommandName(SymfonyStyle $io): string
+    {
+        do {
+            $commandName = (string)$io->ask(
+                'Please provide the command name (example: "ext:clean")',
+            );
+
+            if ($commandName === '') {
+                $io->error('Command name can not be empty.');
+                $validCommandName = false;
+            } elseif (preg_match('/^\d/', $commandName)) {
+                $io->error('Command name should not start with a number.');
+                $validCommandName = false;
+            } elseif (preg_match('/[^a-zA-Z0-9:]/', $commandName)) {
+                $io->error('Command name contains invalid chars. Please provide just letters, numbers and colon (:).');
+                $validCommandName = false;
+            } else {
+                $validCommandName = true;
+            }
+        } while (!$validCommandName);
+
+        return $commandName;
+    }
+
+    private function askForCommandAliases(SymfonyStyle $io): array
+    {
+        $commandAliases = (string)$io->ask(
+            'Provide an alias name for your command',
+        );
+
+        if ($commandAliases === '') {
+            return [];
+        }
+
+        return [$commandAliases];
+    }
+
+    private function tryToCorrectCommandClassName(string $givenCommandClassName): string
```

Comment:
This method is all over the place, we should centralise it once and reuse it with the prefix as parameter. But that can be done in a follow up

Reply:
Correct. Perfect for a trait. No additional class properties, const or methods needed

Reply:
Should be streamlined in a follow up. I think an injectable Service would be better suited but let us carry discussion elsewhere